### DISCUSSION TITLE
V3: Javascript dosn't work after using wire:navigate

### DISCUSSION THREAD

**Discussion by denandraramadhan** [livewire/livewire#5838] [7. Bugs (archived)] [answered]
I will try to explain the specifics of my problem. I am using AdminLTE Template, where I put the js file in the public folder. Then I call it on the view layout with `<script src="{{ asset('js/app.js')}}"></script>` Where the script will bring up the icon and toogle (hide/show) sidebar. For the first visit the page is normal, but when I visit another menu with the same layout, then the script is not working, the icon is not there and the toggle is not working. Even though I see in the console / element that the script is rendered, but somehow the script does not work when not reloading the page or when using `wire:navigate`.
Thanks in advance

![image](https://github.com/livewire/livewire/assets/81479948/9a265840-1bc3-4b8e-bcc8-6ad8680f7340) ![image](https://github.com/livewire/livewire/assets/81479948/32a33780-067c-406d-8266-1db002102694)

**Comment by joshhanley** [comment]
@denikurnia11 can you share a bit more code with what your layout looks like and where you are including the script?

**Reply by denandraramadhan** [reply]
I put the script in the body tag. And on the nav (sidebar) there is an icon that is rendered through the script, but after switching pages, the script seems unused. But the script is rendered after I look at the network

![code](https://github.com/livewire/livewire/assets/81479948/6c4afe4c-6542-4b38-902d-7af74c2a54c0)

**Comment by joshhanley** [comment]
@denikurnia11 as your script tag should be available on all pages, I would suggest wrapping it in the `@persist` tag so it doesn't get reloaded. See here for details https://livewire.laravel.com/docs/navigate#persisting-elements-across-page-visits

Hope this helps!

**Reply by joshhanley** [reply]
@denikurnia11 sorry, it should actually be put `data-navigate-once` on the script tag, see here https://livewire.laravel.com/docs/navigate#scripts-in-the-body-are-re-evaluated

**Reply by cuzinxyz** [reply]
Sorry for waking up this answer, the above problem i am also facing, the ideas you gave are not working.
Hope to receive comments from you.

**Reply by Instartec** [reply]
I think you dont understand whats happening with him, the same happens to me, when i use wire:navigate my JS scripts in body tag, like bootstrap.min.js, stop working, it seems to be a severe bug.

**Reply by denandraramadhan** [reply, marked as answer]
you can find the answer here
https://youtu.be/gkDq2E6g0F0?si=iLGNzMtkgtIdxZ8f
